### ISSUE TITLE
Display of nodes for hierachical list and selections (transfer from Knora repository)

### ISSUE THREAD

**Issue by mrivoal** [dhlab-basel/Salsah#2]
_Following @sofiag's suggestion, I transfer the issue previously opened on Knora repository as issue #321 here. It is a possible requirement for Salsah 2_

I am wondering whether the **current default display for selections' and hierarchical lists' nodes** is the most convenient.

![capture d ecran 2016-11-11 a 11 24 03 copie](https://cloud.githubusercontent.com/assets/17294309/20219191/86302566-a828-11e6-8399-6dda31791508.png)

It is my understanding that, for **selections**, the first node of the list is selected by default. 

For **hlist**, the first node of the list is also selected by default, but the first subnode is left blank. I assume it is so in order to let the user select the first node without going further if the 1st level of the hierarchy is where he wants to stop.

If the user doesn't want any value for a property with a `knora-base:objectClassConstraint` on `knora-base:ListValue` , he will have to delete the field by clicking on the red cross because he can't leave the first entry for a list blank . But **properties for this value do not behave like others**: if you don't want any value to be entered for a `knora-base:TextValue`, then you can just leave it blank and create the resource as is (provided that the cardinality allows it). 

My concern is about the errors the current display could result in for users. **Shouldn't the behaviour for all the values be the same?** What about displaying the first node of a list as blank so that users can simply leave it like this if nothing is to be entered in this field instead of having him deleting it?

**Comment by lrosenth** [comment]
Yes Marion, Your right – not very consistent indeed! I should change this as suggested…. I do not have the code in mind (if there is a “technical” reason such as lazyness of the programmer to it… ;-)


Lukas

P.S. Assign an issue to me!

--  

Prof. Dr. Lukas Rosenthaler
Head of "Digital Humanities Lab",
University of Basel
Bernoullistrasse 32
CH-4056 Basel, Switzerland

Tel: + 41 61 267 0490
Fax: +41 61 267 3855
http://www.dhlab.unibas.ch
